Kernel/Arch/aarch64/Interrupts.cpp

```diff
@@ -70,8 +70,7 @@ static PageFault page_fault_from_exception_syndrome_register(VirtualAddress faul
 
     fault.set_access((esr_el1.ISS & (1 << 6)) == (1 << 6) ? PageFault::Access::Write : PageFault::Access::Read);
 
-    // FIXME: Set correct mode
-    fault.set_mode(ExecutionMode::Kernel);
+    fault.set_mode(esr_el1.EC == 0x20 || esr_el1.EC == 0x24 ? ExecutionMode::User : ExecutionMode::Kernel);
```

Comment:
nit: might be nice to have some method with a descriptive name on `esr_el1` that we'd call here instead of these magic numbers (and the method would do `return EC == 0x20 || EC == 0x24;`)

Reply:
[the docs](https://developer.arm.com/documentation/ddi0601/2022-12/AArch64-Registers/ESR-EL1--Exception-Syndrome-Register--EL1-?lang=en) are pretty lengthy here. i think that it should probably be two methods returning each of the alternative subclause.

those would be “Instruction Abort from a lower Exception level” and “Data Abort exception from a lower Exception level” which seems to be a reasonable level of details here (suggested names: `isInstructionAbortFromEL0` and `isDataAbortFromEL0`?).

of course those could be merged but i feel that having the control here instead of registers would be way better.

Reply:
Added a helper called `exception_class_is_data_or_instruction_abort_from_lower_exception_level` and used it instead.